### PR TITLE
Adding option to control behaviour of heat of formation

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
A couple of people have noted that the base form of the Shomate equation used by NIST subtracts the heat of formation from the pure component enthalpy calculation. This creates issues in systems with reactions or phase equilibria where the latent heat is not accounted for.

To address this, a new configuration option has been added to the generic property framework which determines whether the heat of formation should be included in the calculation of specific enthalpy, and this option applied in all pure component property methods to determine whether the heat of formation is included (or in the case of NIST, subtracted).

## Changes proposed in this PR:
- Adds new `include_enthalpy_of_formation` configuration argument
- Updates all pure component property libraries to make use of this argument to determine whether heat of formation is required.
- Added some documentation and tests.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/user_guide/components/property_package/general/pure/NIST.rst [hunk 1]
```diff
@@ -58,7 +58,7 @@ Units are :math:`\text{J/mol}`.
    ":math:`H`", "cp_mol_ig_comp_coeff_H", ":math:`\text{kJ/mol}`", ""
 
 .. note::
-    This correlation uses the same parameters as for the ideal gas heat capacity with additional parameters `F` and `H`. These parameters account for the enthalpy at the reference state defined by NIST, where `F` is the constant of integration and `H` is the standard molar heat of formation. Note that the default form of the expression used by NIST subtracts the heat of formation from the specific enthalpy.
+    This correlation uses the same parameters as for the ideal gas heat capacity with additional parameters `F` and `H`. These parameters account for the enthalpy at the reference state defined by NIST, where `F` is the constant of integration and `H` is the standard molar heat of formation. Note that the default form of the expression used by NIST subtracts the heat of formation from the specific enthalpy. This behavior can be controlled using the global configuration argument `include_enthalpy_of_formation` - if this is set to `True` (the default setting), then the `H` term is not used when calculating secific enthalpies.
```
Comment (agarciadiego) [inline docs/user_guide/components/property_package/general/pure/NIST.rst:61]: specific enthalpies

**Review by agarciadiego — APPROVED** [review]
Just a typo otherwise LGTM

File: docs/user_guide/components/property_package/general/pure/NIST.rst (hunk 1 above)
Comment (andrewlee94) [reply docs/user_guide/components/property_package/general/pure/NIST.rst]: Fixed.

**Review by agarciadiego — APPROVED** [review]
LGTM

**Review by jghouse88 — APPROVED** [review]
LGTM

**Comment by andrewlee94** [comment]
@agarciadiego You should be good to do your changes now.

**Pull request merged**